distribution/registry.go

```diff
@@ -94,7 +95,12 @@ func NewV2Repository(repoInfo *registry.RepositoryInfo, endpoint registry.APIEnd
 	creds := dumbCredentialStore{auth: authConfig}
```

Comment:
Can we avoid creating these objects if they are not needed now?

Or even may be more readable if this just sets up array of handlers and there is only a single call to `NewAuthorizer`.